bin/tests/cli.rs

```diff
@@ -215,6 +214,37 @@ fn test_truncate_file() {
     agent_handle.kill().expect("Could not kill process");
 }
 
+#[test]
+#[cfg_attr(not(feature = "integration_tests"), ignore)]
+fn test_exclusion_rules() {
+    let dir = tempdir().expect("Could not create temp dir").into_path();
+    let included_file = dir.join("file1.log");
+    let excluded_file = dir.join("file2.log");
+    common::append_to_file(&included_file, 100, 50).expect("Could not append");
+    common::append_to_file(&excluded_file, 100, 50).expect("Could not append");
+
+    let mut agent_handle = common::spawn_agent(AgentSettings {
+        log_dirs: &dir.to_str().unwrap(),
+        exclusion_regex: r"\w+ile2\.\w{3}",
+    });
+
+    let mut stderr_reader = BufReader::new(agent_handle.stderr.as_mut().unwrap());
+
+    let lines = common::wait_for_file_event("initialized", &included_file, &mut stderr_reader);
+    let re = Regex::new(r"initialized [^\n]*file2\.log").unwrap();
```

Comment:
One other minor thing is that the predicate crate has Regex predicates, ie
https://docs.rs/predicates/1.0.6/predicates/str/struct.RegexPredicate.html which would save the additional crate import, but again very minor.